tests/browsercontext-route.spec.ts

```diff
@@ -212,3 +212,66 @@ it('should support the times parameter with route matching', async ({ context, p
   await page.goto(server.EMPTY_PAGE);
   expect(intercepted).toHaveLength(1);
 });
+
+it(`should overwrite post body`, async ({ context, server, page }) => {
+  server.setRoute('/load', (req, res) => {
+    res.write(`
+      <script>
+        (async () => {
+          await fetch('/empty.html', {
+            method: 'POST',
+            body: 'original',
+          });
+        })()
+      </script>
+    `);
+    res.end();
+  });
+
+  await context.route('**/empty.html', route => {
+    route.continue({
+      postData: 'replaced',
+    });
+  });
+
+  const [req] = await Promise.all([
+    server.waitForRequest('/empty.html'),
+    page.goto(server.PREFIX + '/load'),
+  ]);
+
+  const body = (await req.postBody).toString();
+  expect(body).not.toBe('original');
+  expect(body).toBe('replaced');
+});
+
+it(`should overwrite post body with empty string`, async ({ context, server, page, browserName }) => {
+  it.fail(browserName === 'firefox', `There's likely a FF-specific bug in Juggler or how we ser/de to Juggler.`);
```

Comment:
@dgozman I don't have an FF checkout currently, so feel free to tackle this one in a follow up MR. It might end up not being Juggler, but my gut says it's in Juggler or code that is unique to talking to Juggler. The empty string is present for this `sendMayFail` call (so I suspect it's after that somewhere):

https://github.com/microsoft/playwright/blob/db0cda9df5c9ee2d28eb2cbeef3e0c04d7abb2a7/packages/playwright-core/src/server/firefox/ffNetworkManager.ts#L216

Reply:
This should now be fixed. I coincidentally fixed while fixing something else.

Reply:
Think I now fixed this. 'twas juggler after all. Patch here: 342a49b6e4bfbc132c6858637ddc7e7570ae6748